### PR TITLE
add pagination to delivery info endpoint

### PR DESCRIPTION
**Pull request by raminismayilov**
Here's how the response to a request with the query parameters 'page=2&size=20' looks like:

image

Pagination is one-indexed, and is implemented using Spring's own resources (PageImpl, Page, Pageable). I extended PageImpl to minimize what Spring serializes as part of its Page object and to establish one-based indexing.

And here's how a serialized message looks  like:

image

It's the serialized form of a MessageContent object, from which various content types (EmailContent, SmsContent, etc) inherit. So if the message is an email, we'll see an email-specific content structure.

I tried to keep my changes as minimal and less intrusive as possible to not cause extensive breaks in tests and elsewhere. Tests required slight modifications to get them back to normal.

### CODE REVIEW

File: src/main/java/com/obj/nc/repositories/DeliveryInfoRepository.java [hunk 1]
```diff
@@ -73,6 +73,42 @@ List<DeliveryInfo> findByIntentIdAndStatusOrderByProcessedOn(
     List<DeliveryInfo> findByEventIdAndEndpointIdOrderByProcessedOn(@Param("eventId") UUID eventId,
                                                                     @Param("endpointId") UUID endpointId);
 
+    String QRY_LATEST_DELIVERY_INFO_BY_ENDPOINT_ID = "WITH temp AS (\n" +
```
Comment (cuzyjan) [inline src/main/java/com/obj/nc/repositories/DeliveryInfoRepository.java:76]: kind of copy/paste this sql and the secong. 
+ 'temp' is not very descriptive name.. would suggest to name the temprary teble/view more related to what it contains

**Review by cuzyjan — CHANGES_REQUESTED** [review]
its just fine apart of that one comment.
I'm thinking if it would make sence to test the pagination by it self.. which means test with page size 1 somethinkg that return at least two delivery infors and try first page and second page.
... BTW: does ReusltPage object has these indication if the FE can fetch next page? - Now i see in you description above: It has

**Comment by cuzyjan** [comment]
I'm thinking about the @class attribute in the message part of our responce.. It is completely ok to have it like this in NC it self but I'm not sure if it is ok to put it into an interface that is used by external systems. It would be better to have somethink more like @Type in the Enpoint class.
This brings me to a second thought. Should we expose our domain entities to the outside world directly like we do? Wouldn't it be sage to have in this case for both Endpoint and MessageBody (in message attribute) a dedicated DTO? hmm... most likely would, the question is if it is worth the efford

**Comment by raminismayilov** [comment]
After adding DTOs, here's how the response from that specific endpoint looks like:

image

image

**Comment by cuzyjan** [comment]
I think the new format is very nice. We got rid of those @, we are internally typ safe, it all seems "object oriented",.. 
Fine for me

**Comment by raminismayilov** [comment]
Filled those missing toDto() methods.

**Review by cuzyjan — APPROVED** [review]

**Pull request merged**